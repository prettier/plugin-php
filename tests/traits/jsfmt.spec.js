run_spec(import.meta, ["php"]);
// see https://github.com/prettier/plugin-php/issues/107
// run_spec(import.meta, ["php"], { openingBraceNewLine: false });
