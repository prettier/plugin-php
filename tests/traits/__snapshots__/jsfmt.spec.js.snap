// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`traits.php 1`] = `
<?php
trait A {
  function testFunction() { /*1*/ }
  function otherTestFunction() { /*2*/ }
}

trait B extends otherTrait {
  function someOtherTestFunction() { /*3*/ }
}

trait C extends A implements B {
  function someOtherTestFunction() { /*4*/ }
  function testFunction() { /*5*/ }
}

class ImplementingClass {
    use testTrait, otherTrait;
    use testTrait, implementingTrait {
      A::testFunction insteadof C;
      B::someOtherTestFunction as aliasedFunctionName;
      C::testFunction as private testVisibility;
    }
}

class ShortImplementingClass {
  use testTrait { A::testFunction insteadof C; }
}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<?php
trait A {
  public function testFunction()
  {
    // 1
  }
  public function otherTestFunction()
  {
    // 2
  }
}

trait B extends otherTrait {
  public function someOtherTestFunction()
  {
    // 3
  }
}

trait C extends A implements B {
  public function someOtherTestFunction()
  {
    // 4
  }
  public function testFunction()
  {
    // 5
  }
}

class ImplementingClass
{

  use testTrait, otherTrait;
  use testTrait, implementingTrait {
    A::testFunction insteadof C;
    B::someOtherTestFunction as aliasedFunctionName;
    C::testFunction as private testVisibility;
  }
}

class ShortImplementingClass
{

  use testTrait { A::testFunction insteadof C; }
}
`;
